httpbp/error_logger.go

```diff
@@ -0,0 +1,80 @@
+package httpbp
+
+import (
+	"context"
+	stdlog "log"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/reddit/baseplate.go/internal/prometheusbpint"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/randbp"
+)
+
+var httpServerLoggingCounter = promauto.With(prometheusbpint.GlobalRegistry).NewCounterVec(prometheus.CounterOpts{
+	Name: "httpbp_server_upstream_issue_logs_total",
+	Help: "Number of logs emitted by stdlib http server regarding an upstream issue",
+}, []string{"upstream_issue"})
+
+type issueConfig struct {
+	rate    float64
+	counter prometheus.Counter
+}
+
+// This is a special zapcore used by stdlib http server to handle error logging.
+type wrappedCore struct {
+	zapcore.Core
+
+	issues map[int64]issueConfig
```

Comment:
Do we need either this to be `map[int64]*issueConfig` or the counter in issueConfig to be a `*prometheus.Counter`? Or is it fine to increment using copies?

Reply:
`prometheus.Counter` is an interface. it's almost never a good idea to get the pointer to an interface.